subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/cunit/CUnitTestResults.groovy

```diff
@@ -27,7 +27,7 @@ class CUnitTestResults {
     CUnitTestResults(TestFile testResultsFile) {
         assert testResultsFile.exists()
         this.testResultsFile = testResultsFile
-        final XmlParser parser = new XmlParser(false, false)
+        final def parser = new XmlParser(false, false)
```

Comment:
You could simplified it further to:
```suggestion
        final parser = new XmlParser(false, false)
```
(twice)

Reply:
Yeah, there's tons of places where we could improve the code, but this PR is just about trying to pass our tests with Groovy 4.

Reply:
Sure, I've just mentioned that one as you changed `XmlParser → def` (and could be `XmlParser → final`), with the same effort.

Reply:
Nevertheless, it's just a nitpicking :).